`targetKey` pointing at a primary key still causes "is not marked as unique" error

### Describe the bug

Make a `p.oneToOne(User).targetKey('id')` where `User.id` is the primary key:
```typescript
orm.discoverEntity(defineEntity({
  name: 'Test',
  properties: {
    id: p.integer().primary().autoincrement(),
    userId: p.manyToOne(User).targetKey('id')
  }
}))
```
Result:
```
Caused by: MetadataError: testModel.nullableReference has 'targetKey' set to 'id', but User.id is not marked as unique
```

### Reproduction

https://github.com/KazimirPodolski/mikro-orm-reproduction/tree/targetKey-vs-primary

### What driver are you using?

@mikro-orm/sqlite

### MikroORM version

7.0.12

### Node.js version

24

### Operating system

_No response_

### Validations

- [x] Read the [Contributing Guidelines](https://github.com/mikro-orm/mikro-orm/blob/master/CONTRIBUTING.md).
- [x] Read the [docs](https://mikro-orm.io/docs).
- [x] Check that there isn't [already an issue](https://github.com/mikro-orm/mikro-orm/issues) that reports the same bug to avoid creating a duplicate.
- [x] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mikro-orm/mikro-orm/discussions) or join our [Discord](https://discord.gg/w8bjxFHS7X).
- [x] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

--- a/packages/core/src/metadata/MetadataValidator.ts
+++ b/packages/core/src/metadata/MetadataValidator.ts
@@ -221,7 +221,7 @@ export class MetadataValidator {
    * Composite unique constraints are not sufficient for targetKey.
    */
   private isPropertyUnique(prop: EntityProperty, meta: EntityMetadata): boolean {
-    if (prop.unique) {
+    if (prop.unique || (prop.primary && meta.primaryKeys.length === 1)) {
       return true;
     }
 

--- a/tests/MetadataValidator.test.ts
+++ b/tests/MetadataValidator.test.ts
@@ -694,6 +694,42 @@ describe('MetadataValidator', () => {
       expect(() => validator.validateEntityDefinition(new MetadataStorage(meta), Book, options)).not.toThrow();
     });
 
+    test('does not throw when targetKey references a primary key', async () => {
+      class Author {}
+      class Book {}
+      const meta = {
+        Author: {
+          name: 'Author',
+          className: 'Author',
+          class: Author,
+          primaryKeys: ['id'],
+          properties: {
+            id: { name: 'id', kind: ReferenceKind.SCALAR, type: 'number', primary: true },
+          },
+        },
+        Book: {
+          name: 'Book',
+          className: 'Book',
+          class: Book,
+          primaryKeys: ['id'],
+          properties: {
+            id: { name: 'id', kind: ReferenceKind.SCALAR, type: 'number', primary: true },
+            author: {
+              name: 'author',
+              kind: ReferenceKind.MANY_TO_ONE,
+              type: 'Author',
+              targetKey: 'id',
+            },
+          },
+        },
+      } as any;
+      meta.Author.root = meta.Author;
+      meta.Book.root = meta.Book;
+      meta.Book.properties.author.targetMeta = meta.Author;
+
+      expect(() => validator.validateEntityDefinition(new MetadataStorage(meta), Book, options)).not.toThrow();
+    });
+
     test('throws when targetKey references property in composite unique index (not sufficient)', async () => {
       class Author {}
       class Book {}

